src/ol/renderer/vector.js

```diff
@@ -289,9 +335,22 @@ function renderMultiPolygonGeometry(builderGroup, geometry, style, feature) {
  * @param {import("../geom/Point.js").default|import("../render/Feature.js").default} geometry Geometry.
  * @param {import("../style/Style.js").default} style Style.
  * @param {import("../Feature.js").FeatureLike} feature Feature.
+ * @param {import("../render/canvas/BuilderGroup.js").default=} opt_declutterBuilderGroup Builder for decluttering.
  */
-function renderPointGeometry(builderGroup, geometry, style, feature) {
+function renderPointGeometry(
+  builderGroup,
+  geometry,
+  style,
+  feature,
+  opt_declutterBuilderGroup
+) {
   const imageStyle = style.getImage();
+  const textStyle = style.getText();
+  let sharedData;
+  if (opt_declutterBuilderGroup) {
+    builderGroup = opt_declutterBuilderGroup;
+    sharedData = imageStyle && textStyle ? {} : undefined;
```

Comment:
For  #11621 to continue to work this should be changed to
```suggestion
    sharedData = imageStyle && textStyle && textStyle.getText() ? {} : undefined;
```

Reply:
I'd prefer if this could be merged before #11621. @openlayers/core - if anyone has a spare cycle, please review this pull request!